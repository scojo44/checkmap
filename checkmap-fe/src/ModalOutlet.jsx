import React from "react"
import {Outlet} from "react-router-dom";
import Modal from 'react-modal'

/** Wrapper to show forms as a modal dialog box */

export default function ModalOutlet(props) {
  // Bind modal to your appElement (https://reactcommunity.org/react-modal/accessibility/)
  Modal.setAppElement('#root');

  const customStyles = {
    overlay: {
      backgroundColor: 'rgba(0,0,0,80%)',
      zIndex: '9999'
    },
    content: {
      top: '10%',
      bottom: 'auto',
      width: '50%',
      margin: 'auto',
      minWidth: '400px',
      borderRadius: '1rem'
    },
  };

  return (
    <Modal
      isOpen={true}
      style={customStyles}
      contentLabel="Modal"
      ariaHideApp={import.meta.env.NODE_ENV !== 'test'} // Suppress appElement warnings during tests
    >
      <Outlet/>
    </Modal>
  );
}
