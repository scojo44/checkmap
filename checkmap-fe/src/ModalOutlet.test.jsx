import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import ReactModal from 'react-modal';
import ModalOutlet from './ModalOutlet'

describe('ModalOutlet Tests', () => {
  // React-Modal expects a top-level element with ID of 'root' for accessibility
  // Adding a <div id="root"> around the <ModalOutlet> doesn't work.
  ReactModal.setAppElement = vi.fn();

  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <ModalOutlet element={<h1>Test</h1>}/>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
          <ModalOutlet element={<h1>Test</h1>}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
