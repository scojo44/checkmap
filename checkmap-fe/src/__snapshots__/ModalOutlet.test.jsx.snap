// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalOutlet Tests > Matches snapshot 1`] = `<DocumentFragment />`;
